### PR TITLE
Support for guild integrations

### PR DESCRIPTION
**Pull request by Soheab**
- [x] GUILD_INTEGRATIONS_UPDATE
- [x] INTEGRATION_CREATE
- [x] INTEGRATION_UPDATE
- [x] INTEGRATION_DELETE

**Note:** This PR does not include support for the relevent audit logs.

### CODE REVIEW

File: discord_http/enums.py [hunk 1]
```diff
@@ -134,6 +134,10 @@ class IntegrationType(BaseEnum):
     guild = 0
     user = 1
 
+class IntegrationExpireBehaviour(BaseEnum):
```
Comment (AlexFlipnote) [inline discord_http/enums.py:137]: This is a duplicate of `enums.ExpireBehaviour`

File: discord_http/integrations.py
Comment (AlexFlipnote) [inline discord_http/integrations.py:1]: Is this gateway spesific or also API?

**Review by AlexFlipnote — CHANGES_REQUESTED** [review]

File: discord_http/enums.py (hunk 1 above)
Comment (Soheab) [reply discord_http/enums.py:137]: Oh, I didn't see that. My bad.

File: discord_http/integrations.py
Comment (Soheab) [reply discord_http/integrations.py:1]: Integrations can be fetched per guild, so both.

**Review by AlexFlipnote — APPROVED** [review]

**Pull request merged**